media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1438,297 @@ sidebar.ArgumentView = class {
+                } else if (bits[newTypeIndex] === 64) {  // float64
+                    view.setFloat64(0, data, true);
+                }
+                for (let j = 0; j < bits[newTypeIndex] / 8; j++) {
+                    newArray.push(view.getUint8(j));
+                }
+            }
+        } else if (typeclass[newTypeIndex] === 2 || typeclass[newTypeIndex] === 1) {  // new type :
+                                                                                      // int or uint
+            for (let i = 0; i < modifiedArr.length; i++) {
+                let data = parseInt(modifiedArr[i]);
+                try {
+                    if (modifiedArr[i] === 'true') {
+                        data = 1;
+                    } else if (modifiedArr[i] === 'false') {
+                        data = 0;
+                    } else {
+                        data = parseInt(modifiedArr[i]);
+                    }
+                } catch (err) {
+                    return;
+                }
+
+                if (typeclass[newTypeIndex] === 1) {  // int
+                    if (bits[newTypeIndex] === 8) {   // int8
+                        view.setInt8(0, data, true);
+                    } else if (bits[newTypeIndex] === 16) {  // int16
+                        view.setInt16(0, data, true);
+                    } else if (bits[newTypeIndex] === 32) {  // int32, int64
+                        view.setInt32(0, data, true);
+                    }
+                } else {  // uint
+                    if (data < 0) {
+                        return;
+                    }
+
+                    if (bits[newTypeIndex] === 8) {  // uint8
+                        view.setUint8(0, data, true);
+                    } else if (bits[newTypeIndex] === 32) {  // uint32
+                        view.setUint32(0, data, true);
+                    } else if (bits[newTypeIndex] === 64) {  // uint64
+                        view.setBigUint64(0, BigInt(parseInt(String(data))), true);
+                    }
+                }
+                for (let j = 0; j < bits[newTypeIndex] / 8; j++) {
+                    newArray.push(view.getUint8(j));
+                }
+            }
+
+        } else if (typeclass[newTypeIndex] === 4) {  // current type : boolean or new type : boolean
+            for (let i = 0; i < modifiedArr.length; i++) {
+                if (modifiedArr[i]) {
+                    newArray.push(1);
+                } else {
+                    newArray.push(0);
+                }
+            }
+        }
+        /* ? -> String */
+        else if (typeclass[newTypeIndex] === 3) {
+            for (let i = 0; i < modifiedArr.length; i++) {
+                var str = String(modifiedArr[i]);
+                for (let j = 0; j < str.length; j++) {
+                    newArray.push(str.charCodeAt(j));
+                }
+            }
+        }
+
+        this._editObject._arguments._initializer._data = newArray;
+
+        return 1;
```

Comment:
Return type is not consistent with others.

Reply:
We are returing null on error cases and returning 1 on normal cases at the moment, but we will change them to 'true's and 'false's then!